libraries/lib-utility/MapToPositiveHalfIndex.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <cassert>
+
+/*!
+ * @brief Useful when dealing with symmetric spectra reduced only to their
+ * positive half. See tests below for more details.
+ * @param fullSize The size of the original vector. Must be strictly positive
+ * and even, or the function will return 0.
+ * @pre fullSize > 0 && fullSize % 2 == 0
+ */
+constexpr auto MapToPositiveHalfIndex(int index, int fullSize)
+{
+   assert(fullSize > 0 && fullSize % 2 == 0);
+   if (index >= 0)
+      index = index % fullSize;
+   else
+      index = fullSize - (-index % fullSize);
+   if (index > fullSize / 2)
+      index = fullSize - index;
+   return index;
+}
+
+static_assert(MapToPositiveHalfIndex(-3, 4) == 1);
```

Comment:
I don't like it, but I don't have a strong opinion, either

Reply:
static-asserting in the header file?

Reply:
Writing unit tests in a header file.